src/Paramore.Brighter/CommandProcessor.cs

```diff
@@ -461,6 +472,166 @@ public async Task PublishAsync<T>(
+            var context = InitRequestContext(span, requestContext);
+
+            try
+            {
+                Message message = s_mediator!.CreateMessageFromRequest(request, context);
+
+                var scheduler = _messageSchedulerFactory.Create(this);
+                return scheduler switch
+                {
+                    IAmAMessageSchedulerSync sync => sync.Schedule(message, at),
+                    IAmAMessageSchedulerAsync async => BrighterAsyncContext.Run(async () => await async.ScheduleAsync(message, at)),
+                    _ => throw new InvalidOperationException("Message scheduler must be sync or async")
+                };
+            }
+            catch (Exception e)
+            {
+                _tracer?.AddExceptionToSpan(span, [e]);
+                throw;
+            }
+            finally
+            {
+                _tracer?.EndSpan(span);
+            }
+        }
+        
+        /// <inheritdoc />
+        public async Task<string> SchedulerPostAsync<TRequest>(TRequest request, 
+            TimeSpan delay,
+            RequestContext? requestContext = null,
+            Dictionary<string, object>? args = null,
+            bool continueOnCapturedContext = true, 
+            CancellationToken cancellationToken = default) where TRequest : class, IRequest
+        {
+            if (_messageSchedulerFactory == null)
+            {
+                throw new InvalidOperationException("No message scheduler factory defined.");
+            }
+            
+            s_logger.LogInformation("Scheduling a request: {RequestType} {Id}", request.GetType(), request.Id);
+            
+            var span = _tracer?.CreateSpan(CommandProcessorSpanOperation.Scheduler, request, requestContext?.Span, options: _instrumentationOptions);
+            var context = InitRequestContext(span, requestContext);
+
+            try
+            {
+                Message message = await s_mediator!.CreateMessageFromRequestAsync(request, context, cancellationToken);
+
+                var scheduler = _messageSchedulerFactory.Create(this);
+                return scheduler switch
+                {
+                    IAmAMessageSchedulerAsync async => await async.ScheduleAsync(message, delay, cancellationToken).ConfigureAwait(continueOnCapturedContext),
+                    IAmAMessageSchedulerSync sync => sync.Schedule(message, delay),
+                    _ => throw new InvalidOperationException("Message scheduler must be sync or async")
+                };
+            }
+            catch (Exception e)
+            {
+                _tracer?.AddExceptionToSpan(span, [e]);
+                throw;
+            }
+            finally
+            {
+                _tracer?.EndSpan(span);
+            }
+        }
+
+        public async Task<string> SchedulerPostAsync<TRequest>(TRequest request, 
```

Comment:
The same comments apply. Overload with an At parameter over a new method convention; only one scheduler type